Return drawable with on resource ready callback

abhriyaroy opened the pull request:
Returning the drawable as a callback will allow the user to use the drawable as any manner which seems fit and not just bound to setting it to an Imageview.

Commit:
User picture drawable as argument in onResourceReady callback

Commit:
Use picture drawable as argument in onResourceReady callback

Commit:
Merge branch 'return-drawable_with-onResourceReady-callback' of github.com:abhriyaroy/GlideToVectorYou into return-drawable_with-onResourceReady-callback

Commit:
Remove imageview argument when using a listener

When using a listener, the drawable obtained in the callback can be used instead of supplying an imageview as an extra argument. This will cater to the cases, where the user only wants to load a PictureDrawable without setting it to any specific imageview.

SubhrajyotiSen commented on library/src/main/java/com/github/twocoffeesoneteam/glidetovectoryou/GlideToVectorYou.java:
```diff
@@ -50,7 +53,7 @@ public GlideToVectorYou setPlaceHolder(int placeHolderLoading, int placeHolderEr
     }
 
-    public void load(Uri uri, ImageView imageView) {
+    public void load(Uri uri) {
```
This will break existing implementations. Maybe add a new method with the method signature that you want?

abhriyaroy replied:
Sure. Kept the older method and just added a new one.

SubhrajyotiSen commented on library/src/main/java/com/github/twocoffeesoneteam/glidetovectoryou/SvgSoftwareLayerSetter.java:
```diff
@@ -41,11 +41,9 @@ public boolean onLoadFailed(GlideException e, Object model, Target<PictureDrawab
     @Override
     public boolean onResourceReady(PictureDrawable resource, Object model,
                                    Target<PictureDrawable> target, DataSource dataSource, boolean isFirstResource) {
-        ImageView view = ((ImageViewTarget<?>) target).getView();
```
You might want to keep this as is for backward compatibility as well

abhriyaroy replied:
Yes. I didn't consider backward compatibility. Have modified the same.

SubhrajyotiSen commented on library/src/main/java/com/github/twocoffeesoneteam/glidetovectoryou/GlideToVectorYouListener.java:
```diff
@@ -1,6 +1,8 @@
 public interface GlideToVectorYouListener {
     void onLoadFailed();
-    void onResourceReady();
+    void onResourceReady(PictureDrawable pictureDrawable);
```
Depending on whether the `PictureDrawable` can be null or not, an annotation here would be helpful. For Kotlin usages

abhriyaroy replied:
Added `@Nullable`

Commit:
Resolve comments on PR

- Add support for backward compaitibility by keeping previous method for normal loading
- Add nullable annotation to onResourceReady
